applications/StructuralMechanicsApplication/custom_processes/set_automated_initial_variable_process.cpp

```diff
@@ -0,0 +1,118 @@
+    KRATOS_ERROR_IF(MathUtils<double>::Norm3(hole_generatrix_axis) < std::numeric_limits<double>::epsilon()) << "The hole generatrix axis has norm zero" << std::endl;
+    
+    const array_1d<double, 3> hole_generatrix_point = mThisParameters["hole_generatrix_point"].GetVector();  
+
+    array_1d<double, 3> normalized_generatrix_vector;   
+    normalized_generatrix_vector[0] = (hole_generatrix_axis[0] * hole_generatrix_axis[0]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[1] = (hole_generatrix_axis[1] * hole_generatrix_axis[1]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+    normalized_generatrix_vector[2] = (hole_generatrix_axis[2] * hole_generatrix_axis[2]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+
+    const double hole_radius_offset = mThisParameters["hole_radius_offset"].GetDouble();
+
+    block_for_each(mrThisModelPart.Elements(), [&](auto& rElement) {
+
+        const array_1d<double, 3>& r_element_centroid = rElement.GetGeometry().Center();
+        
+        array_1d<double, 3> relative_position_vector;
+        relative_position_vector = r_element_centroid - hole_generatrix_point;
+
+        double vector_scaler = relative_position_vector[0] * normalized_generatrix_vector[0] + relative_position_vector[1] * normalized_generatrix_vector[1] + relative_position_vector[2] * normalized_generatrix_vector[2];
+        
+        const array_1d<double, 3> intersection_point = hole_generatrix_point + vector_scaler * normalized_generatrix_vector;
+
+        const array_1d<double, 3> radial_position_vector = r_element_centroid - intersection_point;
+
+        double centroid_relative_distance = std::sqrt(radial_position_vector[0] * radial_position_vector[0] + radial_position_vector[1] * radial_position_vector[1] + radial_position_vector[2] * radial_position_vector[2]) - hole_radius_offset;
+
+        if (centroid_relative_distance < 0.0){
+            if (std::abs(centroid_relative_distance) <= tolerance) {
+                centroid_relative_distance = 0.0;
+            }
+            else {
+                const int elem_id = rElement.Id();
+                KRATOS_ERROR << "Thickness of element " << elem_id << " is too small." << std::endl;
+            }
+        }
+        
+        array_1d<double, 6> initial_variable_vector;
+        noalias(initial_variable_vector) = ZeroVector(6); 
+ 
+        Vector table_id_vector = mThisParameters["table_id_vector"].GetVector();
+
+        for (IndexType count = 0; count < table_id_vector.size(); ++count) {
+                int table_id = table_id_vector[count];
```

Comment:
for efficiency, can you declare the` int` outside the `for` and then change its value inside?